File: NetFramework/Charts/VB/Function2D/Engine.vb

Imports System
Imports System.Drawing



Public Class Engine
   Private func As New FunctionEvaluator()
   
   Public Sub New()
   End Sub 'New
   
   
   Public Function Run([text] As String, xmin As Single, xmax As Single, npts As Integer) As PointF()
      func.Text = [text]
      
      If Not func.Compile() Then
         Return Nothing
      End If 
      Dim data As PointF() = CType(Array.CreateInstance(GetType(PointF), npts), PointF())
      Dim dx As Single = (xmax - xmin) /(npts - 1)
      Dim i As Integer
      For i = 0 To npts - 1
         data(i).X = xmin + dx * i
         data(i).Y = CSng(func.Invoke(data(i).X))
      Next i
      
      Return data
   End Function 'Run
   
   
   Public ReadOnly Property Errors() As String()
      Get
         Return func.Errors
      End Get
   End Property
   
   
   Public Function Compile([text] As String) As Boolean
      func.Text = [text]
      
      Return func.Compile()
   End Function 'Compile
End Class 'Engine

File: NetFramework/Charts/CS/function2d/Engine.cs
using System;
using System.Drawing;

namespace Function2D
{
		public class Engine
		{
			FunctionEvaluator func = new FunctionEvaluator();
			public Engine()
			{
			}

			public PointF[] Run( string text, float xmin, float xmax, int npts)
			{
				func.Text = text;

				if(!func.Compile())
					return null;

				PointF[] data = (PointF[])Array.CreateInstance( typeof(PointF), npts);
				float dx = (xmax-xmin)/(npts-1);
				for( int i=0; i<npts; i++)
				{
					data[i].X = xmin + dx*i;
					data[i].Y = (float)func.Invoke( data[i].X);
				}

				return data;
			}

			public string[] Errors
			{
				get
				{
					return func.Errors;
				}
			}

			public bool Compile( string text)
			{
				func.Text = text;

				return func.Compile();
			}
		}
	}
